README.md

```diff
@@ -1,3 +1,91 @@
 # swift-extras-uuid
 
-A reimplementation of UUID in Swift without the use of Foundation. 
+A reimplementation of UUID in Swift without the use of Foundation or any other dependency. 
+
+[![Swift 5.1](https://img.shields.io/badge/Swift-5.1-blue.svg)](https://swift.org/download/)
+[![github-actions](https://github.com/fabianfett/swift-extras-uuid/workflows/CI/badge.svg)](https://github.com/fabianfett/swift-extras-uuid/actions)
+[![codecov](https://codecov.io/gh/fabianfett/swift-extras-uuid/branch/main/graph/badge.svg)](https://codecov.io/gh/fabianfett/swift-extras-uuid)
+![macOS](https://img.shields.io/badge/os-macOS-green.svg?style=flat)
+![tuxOS](https://img.shields.io/badge/os-tuxOS-green.svg?style=flat)
+
+### Motivation
+
+Foundation on non Darwin platforms (Linux, Windows) has often worse performance and/or missing features compared to its Darwin counterpart. That's why I like to limit the use of Foundation classes and structs to the absolute minimum when writing server side or cross platform Swift code. This package offers the same UUID functionality that Foundation provides, without actually using Foundation. The performance is slightly better on Linux compared to the Foundation implementation and comparable on macOS. 
+
+### How to use it?
+
+Add dependency to your `Package.swift`.
+
+```swift
+  dependencies: [
+    .package(url: "https://github.com/fabianfett/swift-extras-uuid.git", .upToNextMajor(from: "0.1.0")),
+  ],
+```
+
+Add `ExtrasUUID` to the target you want to use it in.
+
+```swift
+  targets: [
+    .target(name: "MyFancyTarget", dependencies: [
+      .product(name: "ExtrasUUID", package: "swift-extras-uuid"),
+    ])
+  ]
+```
+
+Import the library and use it. In case you import Foundation in the same file, this package's UUID is prefixed with an X to avoid naming conflicts.
```

Comment:
```suggestion
Import the library and use it. This package's UUID is prefixed with an X to avoid naming conflicts in cases where you import Foundation in the same file.
```

Slight restructuring, sounded to me a bit like the `UUID` type is only prefixed with an X in cases where Foundation is imported, might be confusing.